python_benchmarks/core.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import ctypes
+import torch
+from torch.profiler import profile, ProfilerActivity
+
+libnames = ("libcuda.so", "libcuda.dylib", "nvcuda.dll", "cuda.dll")
+for libname in libnames:
+    try:
+        cuda = ctypes.CDLL(libname)
+    except OSError:
+        continue
+    else:
+        break
+else:
+    raise OSError("could not load any of: " + " ".join(libnames))
+
+CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE = 38
+l2_cache_size = ctypes.c_int()
+device = torch.cuda.current_device()
+cuda.cuDeviceGetAttribute(
+    ctypes.byref(l2_cache_size), CU_DEVICE_ATTRIBUTE_L2_CACHE_SIZE, device
+)
+
+L2_CACHE_SIZE = l2_cache_size.value
+
+
+def clearL2Cache():
```

Comment:
How about https://pytorch.org/docs/stable/generated/torch.cuda.empty_cache.html?

Reply:
Also, regarding code style, this should be:
```suggestion
def clear_l2_cache():
```
See: https://peps.python.org/pep-0008/#function-and-variable-names
> Function names should be lowercase, with words separated by underscores as necessary to improve readability.

Reply:
> How about https://pytorch.org/docs/stable/generated/torch.cuda.empty_cache.html?

I think `torch.empty_cache` does different thing from `clearL2Cache`. `torch.empty_cache` is mostly to work with PyTorch's CUDA caching allocator, which manages a pool for CUDA memory to avoid `cudaMalloc` and `cudaFree`, and `torch.empty_cache` tells the allocator to `cudaFree` all the memories in its pool.

Reply:
Yes. `clear_l2_cache` is intended to flush the inputs out of L2 cache before benchmarking. This is different from `torch.empty_cache` which seems to only release unoccupied cached memory.